### ISSUE TITLE
JDialog display size difference between 1.8.0_275 and 1.8.0_282 32bits

### ISSUE THREAD

**Issue by SSebUK** [corretto/corretto-8#295]
### Describe the bug
JDialog display size is different between 275 and 282 when using setResizable(false).
282 displays smaller dialogs.
This happens on Windows 10, with Amazon Corretto 1.8.0_282 32 bits, but not for instance with RedHat JRE with same version, so I don't think it's related to OpenJDK itself.

### To Reproduce
This code creates a Frame, and a button in the frame.
When clicking on the button, the dialog is being displayed.
Run one instance of this program with 1.8.0_275 32bits, then another one with 1.8.0_282 32bits and move the dialogs side by side to see the size difference (width and height)

```
import java.awt.Dimension;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

import javax.swing.JButton;
import javax.swing.JDialog;
import javax.swing.JFrame;
import javax.swing.JLabel;
import javax.swing.JPanel;

public class JDialogTestClass extends JFrame implements ActionListener {

    // frame 
    static JFrame f; 
  
    // main class 
    public static void main(String[] args) 
    { 
      // create a new frame 
        f = new JFrame("frame"); 
  
        // create a object 
        JDialogTestClass s = new JDialogTestClass(); 
  
        // create a panel 
        JPanel p = new JPanel(); 
  
        JButton b = new JButton("click"); 
  
        // add actionlistener to button 
        b.addActionListener(s); 
  
        // add button to panel 
        p.add(b); 
  
        f.add(p); 
  
        // set the size of frame 
        f.setSize(700, 500); 
  
        f.setVisible(true); 
    }
    
    public void actionPerformed(ActionEvent e) 
    { 
        String s = e.getActionCommand(); 
        if (s.equals("click")) { 
            // create a dialog Box 
            JDialog d = new JDialog(f, "dialog Box"); 
          
            // NOT resizable
            d.setResizable(false);
            
            // Set fixed dimension
            d.setMinimumSize(new Dimension(320, 420));
            d.setPreferredSize(new Dimension(320, 420));
            d.setMaximumSize(new Dimension(320, 420));
  
            // create a label 
            JLabel l = new JLabel("Reported dim: "+d.getWidth() + "x"+ d.getHeight()); 
  
            d.add(l); 
  
            // set visibility of dialog 
            d.setVisible(true); 
        } 
    } 
  }
```

### Expected behavior
The dialog sizes should be the same between these 2 versions of JRE

### Screenshots
Width difference (282 at the top):
![image](https://user-images.githubusercontent.com/80828256/111466276-fc7ecd00-871a-11eb-8b32-d2fe5d6335f7.png)

Height difference (282 on the left):
![image](https://user-images.githubusercontent.com/80828256/111466382-1fa97c80-871b-11eb-9507-422b327c19c6.png)

### Platform information
    OS: Windows 10 64bits
    Version Amazon Corretto 1.8.0_282-b08

### Additional context
I didn't test with JRE 64bits, as our commercial application HAS to run on 32bits as well. I didn't try on other Windows versions.
This makes a big difference to our app as some components are truncated, rendering them difficult to use.

**Comment by benty-amzn** [comment]
Hi, thanks for contacting us about this. I've confirmed that your provided code does produce windows of different sizes on Windows 10 for both x86 and x64 Corretto JDKs.

**Comment by benty-amzn** [comment]
This appears to be the result of a difference in toolchains between Corretto 8u275 and Corretto 8u282

Corretto 8u282 was [updated away from the Visual Studio 2010 toolchain](https://github.com/corretto/corretto-8/commit/c34e5b8c94348edab3c0ada26509bc7baf252daa) which is no longer supported.
One difference resulting from this change is the version of `msvcr.dll` and `msvcp.dll` bundled - [8u282](https://github.com/corretto/corretto-8/blob/release-8.282.08.1/installers/windows/zip/build.gradle#L38-L39) vs [8u275](https://github.com/corretto/corretto-8/blob/release-8.275.01.1/installers/windows/zip/build.gradle#L38)
This appears to impact the behavior of the Windows APIs underlying Swing and AWT, and the change is present in all newer versions of Visual Studio.

Our investigation has determined that the actual size of the produced dialog on Windows 10 is not `320x420` in either case. Differing assumptions are made by the bundled Microsoft Visual C++ Runtime about the size of the border which will be rendered on the window, which are included in the reported width and height of the window.
This results in a usable size, without borders, of either `306x413` (The new behavior, which assumes Windows 8 style borders, which are quite thick) or `316x418` (the old behavior, which assumes Windows 7 style borders, which are slimmer than Windows 8 but not as slim as Windows 10). 
In both cases, the API reports the size of the window to the JDK as `320x420`.

As a test, we ran the provided code on Windows Server 2012, which uses the same graphical style as Windows 8, in EC2 using Corretto 8u282. The below image has total dimensions of `320x420`, including borders, title bar, etc.
![image](https://user-images.githubusercontent.com/61255184/111709947-dbdf7180-8805-11eb-8e36-3c6378839f0a.png)

Additionally, we have confirmed that the same change in behavior is present with other open source JDK distributions using similar toolchains, as well as Oracle JDK.

Since the Visual Studio 2010 toolchain is no longer supported, it is not feasible to revert this change. 
Our best suggestions are to switch to a dynamic layout, consider enabling window resizing, or update your constant sizes to a larger value which takes the possibility of window decorations into account.
